app/design/frontend/base/default/template/boltpay/catalog/product/button.phtml

```diff
@@ -15,8 +15,39 @@
  * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
 
+
 /* @var $this Bolt_Boltpay_Block_Catalog_Product_Boltpay */
 ?>
-<?php if ($this->isEnabledProductPageCheckout() && $this->isSupportedProductType()): ?>
+<?php if ($this->isEnabledProductPageCheckout()): ?>
+    <!-- Bolt need this for all cases of PPC -->
+    <style>
+<!--
+.disabledbutton {
+    pointer-events: none;
+    opacity: 0.4;
+}
+-->
+</style>
     <div class="bolt-product-checkout-button"></div>
+    <?php if ($this->isSupportedProductType()):?>
+    <?php $token = $this->getBoltToken(new Bolt_Boltpay_Model_BoltOrder(), new Bolt_Boltpay_Helper_CatalogHelper());?>
+    <?php if ($token):?>
+        <script type="text/javascript">
+        jQuery(document).ready(function() {
+                var cart = {
+                    "orderToken": "<?php echo $token;?>"
+                };
+                var hints = {};
+                var callbacks = {};
+                BoltCheckout.configureProductCheckout(cart, hints, callbacks);
```

Comment:
where did you find documentation for this? you should pass cart information not token here https://docs.bolt.com/docs/54-product-checkout

idea is we don't create token until user click bolt checkout button

Reply:
@daisy1754 this is for simple product type.

Reply:
@daisy1754 for different types of product we could create token or we need to validate product.
This is for simple type product